File: app/src/main/java/com/example/shortsgenerator/MainActivity.java
      package com.example.shortsgenerator;
import android.widget.Spinner;
import android.os.Bundle;
import android.widget.*;
import android.content.*;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.FileProvider;

import java.io.File;

import com.example.shortsgenerator.logic.VideoPlan;

public class MainActivity extends AppCompatActivity {
private String formatTime(int seconds) {
    int h = seconds / 3600;
    int m = (seconds % 3600) / 60;
    int s = seconds % 60;

    return String.format("%02d:%02d:%02d,000", h, m, s);
}
    private VideoPlan videoPlan;   // ✅ ОДИН экземпляр
    private File lastSrtFile;
private File generateSrt(String text) {
    try {
        File srtFile = new File(getFilesDir(), "subtitles.srt");
        StringBuilder srt = new StringBuilder();

        String[] lines = text.split("[.!?]\\s*");

        int index = 1;
        int startTime = 0;

        for (String line : lines) {
            int endTime = startTime + 2;

            srt.append(index).append("\n");
            srt.append(formatTime(startTime))
               .append(" --> ")
               .append(formatTime(endTime))
               .append("\n");
            srt.append(line.trim()).append("\n\n");

            startTime = endTime;
            index++;
        }

        try (FileOutputStream fos = new FileOutputStream(srtFile)) {
            fos.write(srt.toString().getBytes());
        }

        return srtFile;

    } catch (Exception e) {
        e.printStackTrace();
        return null;
    }
}
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
srtButton.setOnClickListener(v -> {
    String text = inputText.getText().toString().trim();

    if (text.isEmpty()) {
        Toast.makeText(this, "Нет текста для SRT", Toast.LENGTH_SHORT).show();
        return;
    }

    lastSrtFile = generateSrt(text);

    if (lastSrtFile != null) {
        Toast.makeText(this, "SRT создан", Toast.LENGTH_SHORT).show();
    } else {
        Toast.makeText(this, "Ошибка создания SRT", Toast.LENGTH_SHORT).show();
    }
});
        EditText inputText = findViewById(R.id.inputText);
        TextView resultText = findViewById(R.id.resultText);
        Button generateButton = findViewById(R.id.btnGenerate);
        Button srtButton = findViewById(R.id.btnSrt);
        Button shareButton = findViewById(R.id.btnShare);
        Spinner speedSpinner = findViewById(R.id.speedSpinner);
        Spinner styleSpinner = findViewById(R.id.styleSpinner);

        videoPlan = new VideoPlan(); // ✅ создаём ОДИН РАЗ

        // speed spinner
        String[] speeds = {"Медленно", "Нормально", "Быстро"};
        speedSpinner.setAdapter(new ArrayAdapter<>(
                this,
                android.R.layout.simple_spinner_item,
                speeds
        ));
         speedAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
speedSpinner.setAdapter(speedAdapter); 

        // style spinner
        String[] styles = {"Классика", "Агрессивный", "Минимал", "TikTok PRO"};
        styleSpinner.setAdapter(new ArrayAdapter<>(
                this,
                android.R.layout.simple_spinner_item,
                styles
        ));
speedAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
speedSpinner.setAdapter(speedAdapter);
        generateButton.setOnClickListener(v -> {

    String text = inputText.getText().toString().trim();
    String speed = speedSpinner.getSelectedItem().toString();
    String style = styleSpinner.getSelectedItem().toString();

    if (text.isEmpty()) {
        resultText.setText("❗ Вставь текст");
        return;
    }

    String result =
            "🎬 Стиль: " + style + "\n" +
            "⏱ Скорость: " + speed + "\n\n" +
            text;

    resultText.setText(result);
});
        shareButton.setOnClickListener(v -> {
            if (lastSrtFile == null || !lastSrtFile.exists()) {
                Toast.makeText(this, "Сначала создай SRT", Toast.LENGTH_SHORT).show();
                return;
            }

            Intent shareIntent = new Intent(Intent.ACTION_SEND);
            shareIntent.setType("application/x-subrip");
            shareIntent.putExtra(
                    Intent.EXTRA_STREAM,
                    FileProvider.getUriForFile(
                            this,
                            getPackageName() + ".provider",
                            lastSrtFile
                    )
            );
            shareIntent.addFlags(Intent.FLAG_GRANT_READ_URI_PERMISSION);
            startActivity(Intent.createChooser(shareIntent, "Поделиться SRT"));
        });
    }
}  

File: app/src/main/java/com/example/shortsgenerator/MainActivity.kt
package com.example.shortsgenerator

import android.content.Intent
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        // SplashScreen (Android 12+)
        installSplashScreen()

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 🔗 View binding (ПОСЛЕ setContentView!)
        val inputText = findViewById<EditText>(R.id.inputText)
        val textCounter = findViewById<TextView>(R.id.textCounter)
        val resultText = findViewById<TextView>(R.id.resultText)

        val btnGenerate = findViewById<Button>(R.id.btnSrt)
        val btnSave = findViewById<Button>(R.id.btnSave)
        val btnShare = findViewById<Button>(R.id.btnShare)

        // 🔢 Счётчик символов
        inputText.addTextChangedListener(object : TextWatcher {
            override fun afterTextChanged(s: Editable?) {
                textCounter.text = "${s?.length ?: 0} символов"
            }
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {}
        })

        // 🎬 Generate SRT
        btnGenerate.setOnClickListener {
            val text = inputText.text.toString().trim()
            if (text.isEmpty()) {
                Toast.makeText(this, "Вставь текст", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            resultText.text = generateSrt(text)
        }

        // 💾 Save
        btnSave.setOnClickListener {
            val text = resultText.text.toString()
            if (text.isBlank()) {
                Toast.makeText(this, "Нет SRT для сохранения", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            saveSrtToFile(text)
        }

        // 📤 Share
        btnShare.setOnClickListener {
            val text = resultText.text.toString()
            if (text.isBlank()) {
                Toast.makeText(this, "Сначала создай SRT", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            val intent = Intent(Intent.ACTION_SEND).apply {
                type = "text/plain"
                putExtra(Intent.EXTRA_TEXT, text)
            }
            startActivity(Intent.createChooser(intent, "Поделиться SRT"))
        }
    }

    // ===== SRT GENERATOR =====
    private fun generateSrt(text: String): String {
        val lines = text.split("\n").map { it.trim() }.filter { it.isNotEmpty() }
        val sb = StringBuilder()
        var startMs = 0
        val durationMs = 1500

        lines.forEachIndexed { index, line ->
            val endMs = startMs + durationMs
            sb.append(index + 1).append("\n")
            sb.append(formatTime(startMs)).append(" --> ").append(formatTime(endMs)).append("\n")
            sb.append(line).append("\n\n")
            startMs = endMs
        }
        return sb.toString()
    }

    private fun formatTime(ms: Int): String {
        val seconds = (ms / 1000) % 60
        val minutes = (ms / 60000)
        val millis = ms % 1000
        return String.format("00:%02d:%02d,%03d", minutes, seconds, millis)
    }

    private fun saveSrtToFile(text: String) {
        try {
            val file = java.io.File(getExternalFilesDir(null), "subtitles_${System.currentTimeMillis()}.srt")
            file.writeText(text)
            Toast.makeText(this, "Сохранено:\n${file.name}", Toast.LENGTH_LONG).show()
        } catch (e: Exception) {
            Toast.makeText(this, "Ошибка сохранения", Toast.LENGTH_SHORT).show()
        }
    }
}
